FireboltNETSDK/Client/FireboltDataReader.cs

```diff
@@ -0,0 +1,473 @@
+        {
+            return Task.FromResult(GetSchemaTable());
+        }
+
+        public override Stream GetStream(int ordinal)
+        {
+            string str = GetString(ordinal);
+            byte[] bytes = new byte[str.Length * sizeof(char)];
+            System.Buffer.BlockCopy(str.ToCharArray(), 0, bytes, 0, bytes.Length);
+            return new MemoryStream(bytes);
+        }
+
+        public override string GetString(int ordinal)
+        {
+            return GetValue(ordinal).ToString() ?? throw new InvalidOperationException($"String representation of column ${ordinal} is null");
+        }
+
+        public override TextReader GetTextReader(int ordinal)
+        {
+            return new StringReader(GetString(ordinal));
+        }
+
+        public override object GetValue(int ordinal)
+        {
+            object value = GetValueSafely(ordinal) ?? throw new InvalidOperationException($"Column ${ordinal} is null");
+            return value;
+        }
+
+
+        public override int GetValues(object[] values)
+        {
+            List<object?> row = _queryResult.Data[_currentRowIndex] ?? new List<object?>();
+            if (row == null)
+            {
+                return 0;
+            }
+            int n = Math.Min(row.Count, values.Length);
+            for (int i = 0; i < n; i++)
+            {
+                values[i] = row[i] ?? throw new InvalidOperationException($"Column ${i} is null");
+            }
+            return n;
+        }
+
+        public override bool IsDBNull(int ordinal)
+        {
+            return GetValueSafely(ordinal) == null;
+        }
+
+        private object? GetValueSafely(int ordinal)
+        {
+            List<object?>? row = _queryResult.Data[_currentRowIndex];
+            if (ordinal < 0 || ordinal > row?.Count - 1)
+            {
+                throw new InvalidOperationException($"Column ${ordinal} does not exist");
+            }
+            if (row == null)
+            {
+                return null;
+            }
+            object? value = row[ordinal];
+            if (value == null)
+            {
+                return null;
+            }
+            var columnType = ColumnType.Of(TypesConverter.GetFullColumnTypeName(_queryResult.Meta[ordinal]));
+            return TypesConverter.ConvertToCSharpVal(value.ToString(), columnType);
+        }
+
+        new public Task<bool> IsDBNullAsync(int ordinal)
+        {
+            return Task.FromResult(IsDBNull(ordinal));
+        }
+
+        public override Task<bool> IsDBNullAsync(int ordinal, CancellationToken cancellationToken)
+        {
+            return Task.FromResult(IsDBNull(ordinal));
+        }
+
+        public override bool NextResult()
+        {
+            throw new FireboltException("Batch operations are not supported");
+        }
+
+        public override Task<bool> NextResultAsync(CancellationToken cancellationToken)
+        {
+            return NextResultAsync();
+        }
```

Comment:
Should those both work the same way, throwing an exception?

Reply:
fixed